Signal/src/ViewControllers/AddToBlockListViewController.m

```diff
@@ -0,0 +1,346 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "AddToBlockListViewController.h"
+#import "CountryCodeViewController.h"
+#import "PhoneNumber.h"
+#import "StringUtil.h"
+#import "UIFont+OWS.h"
+#import "UIUtil.h"
+#import "UIView+OWS.h"
+#import "ViewControllerUtils.h"
+#import <SignalServiceKit/PhoneNumberUtil.h>
+#import <SignalServiceKit/TSBlockingManager.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+NSString * const kAddToBlockListViewControllerCellIdentifier = @"kAddToBlockListViewControllerCellIdentifier";
+
+#pragma mark -
+
+// TODO: Add a list of contacts to make it easier to block contacts.
+@interface AddToBlockListViewController () <CountryCodeViewControllerDelegate, UITextFieldDelegate>
+
+@property (nonatomic, readonly) TSBlockingManager *blockingManager;
+
+@property (nonatomic) UIButton *countryNameButton;
+@property (nonatomic) UIButton *countryCodeButton;
+
+@property (nonatomic) UITextField *phoneNumberTextField;
+
+@property (nonatomic) UIButton *blockButton;
+
+@property (nonatomic) NSString *callingCode;
+
+@end
+
+#pragma mark -
+
+@implementation AddToBlockListViewController
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    [self.navigationController.navigationBar setTranslucent:NO];
+}
+
+- (void)loadView
+{
+    [super loadView];
+   
+    self.view.backgroundColor = [UIColor whiteColor];
+    
+    _blockingManager = [TSBlockingManager sharedManager];
+
+    self.title = NSLocalizedString(@"SETTINGS_ADD_TO_BLOCK_LIST_TITLE", @"");
+
+    [self createViews];
+    
+    [self populateDefaultCountryNameAndCode];
+
+    [self addNotificationListeners];
+}
+
+- (void)addNotificationListeners
+{
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(blockedPhoneNumbersDidChange:)
+                                                 name:kNSNotificationName_BlockedPhoneNumbersDidChange
+                                               object:nil];
+}
+
+- (void)dealloc
+{
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (void)createViews {
+    // Country Row
+    UIView *countryRow = [self createRowWithHeight:60 previousRow:nil];
+
+    _countryNameButton = [UIButton buttonWithType:UIButtonTypeCustom];
+    _countryNameButton.titleLabel.font = [UIFont ows_mediumFontWithSize:16.f];
+    [_countryNameButton setTitleColor:[UIColor blackColor]
+                             forState:UIControlStateNormal];
+    [_countryNameButton setTitle:NSLocalizedString(@"REGISTRATION_DEFAULT_COUNTRY_NAME", @"")
```

Comment:
Done. I've added comments for all of the l10n text in this file - and ensured that other usages of these strings have the same comments.